### PR TITLE
impl(bigquery): build changes for the new bigquery v2 target

### PR DESCRIPTION
**Pull request by jsrinnn**
Added new target for Bigquery V2 apis. 

This PR includes the build changes for bazel and CMake. Cmake files to begin with do not include the install instructions. Just the compile and test changes.

I have included the first V2 response header file with rest dependency to show that the build changes are working. Included stubbed out implementation and test files as well. Once this PR is merged I will include the rest of the V2 header files and stubbed out impl and unit tests. This is being done to limit the PR size.

Ran the following to verify everything is building correctly:

// format check
ci/cloudbuild/build.sh -t checkers-pr

// Make sure  CMAKE is successful
cmake -D GOOGLE_CLOUD_CPP_ENABLE=bigquery_v2_minimal -G Ninja -S . -B build-out/home -DCMAKE_TOOLCHAIN_FILE="${VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake"
cmake  --build build-out/home

// Make sure bazel is successful
bazel build //...

// Run project unit tests
env -C build-out/home ctest --output-on-failure -LE integration-test

// Run clag-tidy-pr
ci/cloudbuild/build.sh -t clang-tidy-pr

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/10824)

### CODE REVIEW

**Review by coryan — DISMISSED** [review]
Please make CMake the source of truth for the file list, and fix the inclusion of `*_test.cc` files into your CMake globs.  The rest is more or less optional and/or we can cleanup later.

File: google/cloud/bigquery/bigquery_v2_minimal_internal.cmake [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ~~~
+
+file(
+    GLOB v2_source_files
+    RELATIVE "${CMAKE_CURRENT_SOURCE_DIR}"
+    "v2/minimal/internal/*.h" "v2/minimal/internal/*.cc")
```
[the remark this answers is not in the document]
Comment (jsrinnn) [reply google/cloud/bigquery/bigquery_v2_minimal_internal.cmake]: Done. Modified it to be explicit list

File: google/cloud/bigquery/CMakeLists.txt [hunk 2]
```diff
@@ -86,6 +85,10 @@ target_include_directories(
 target_compile_options(google_cloud_cpp_bigquery_mocks
                        INTERFACE ${GOOGLE_CLOUD_CPP_EXCEPTIONS_FLAG})
 
+# Include bigquery_v2_minimal target for rest transport. TODO : Add ability to
```
[the remark this answers is not in the document]
Comment (jsrinnn) [reply google/cloud/bigquery/CMakeLists.txt]: Removed all redundant TODOs for implement as its obvious these are stubbed out.

File: google/cloud/bigquery/v2/minimal/internal/bigquery_http_response_test.cc [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+#include "google/cloud/bigquery/v2/minimal/internal/bigquery_http_response.h"
+#include <gmock/gmock.h>
+
+namespace google {
+namespace cloud {
+namespace bigquery_v2_minimal_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+TEST(BigQueryHttpResponseTest, BuildFromRestResponseSuccess) {
+  // TODO(jsrinnn): Implement.
```
[the remark this answers is not in the document]
Comment (jsrinnn) [reply google/cloud/bigquery/v2/minimal/internal/bigquery_http_response_test.cc]: Removed TODO.

File: cmake/GoogleCloudCppFeatures.cmake [hunk 4]
```diff
@@ -227,6 +228,10 @@ function (google_cloud_cpp_enable_features)
             if (NOT ("storage" IN_LIST GOOGLE_CLOUD_CPP_ENABLE))
                 add_subdirectory(google/cloud/storage)
             endif ()
+        elseif ("${feature}" STREQUAL "bigquery_v2_minimal")
```
[the remark this answers is not in the document]
Comment (jsrinnn) [reply cmake/GoogleCloudCppFeatures.cmake:231]: Ack.

File: google/cloud/bigquery/v2/minimal/internal/bigquery_http_response.h [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  std::unordered_map<std::string, std::string> http_headers;
+  std::string payload;
+};
+
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_END
+}  // namespace bigquery_v2_minimal_internal
+}  // namespace cloud
+}  // namespace google
+
+#endif  // GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_BIGQUERY_V2_MINIMAL_INTERNAL_BIGQUERY_HTTP_RESPONSE_H
```
[the remark this answers is not in the document]
Comment (jsrinnn) [reply google/cloud/bigquery/v2/minimal/internal/bigquery_http_response.h]: Done. Fixed

File: google/cloud/bigquery/v2/minimal/internal/bigquery_http_response.cc [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+namespace bigquery_v2_minimal_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+StatusOr<BigQueryHttpResponse> BigQueryHttpResponse::BuildFromRestResponse(
+    rest_internal::RestResponse& rest_response) {
+  BigQueryHttpResponse response;
+  auto payload =
+      rest_internal::ReadAll(std::move(rest_response).ExtractPayload());
+  if (!payload.ok()) return payload.status();
+  // TODO(jsrinnn): Implement.
```
[the remark this answers is not in the document]
Comment (jsrinnn) [reply google/cloud/bigquery/v2/minimal/internal/bigquery_http_response.cc]: Removed TODO

File: google/cloud/bigquery/BUILD.bazel [hunk 7]
```diff
@@ -30,15 +29,39 @@ filegroup(
...
         "*.h",
         "internal/*.h",
-        "v2/minimal/internal/*.h",
     ]),
 )
 
+filegroup(
+    name = "v2_srcs",
+    srcs = [
+        "v2/minimal/internal/bigquery_http_response.cc",
```
[the remark this answers is not in the document]
Comment (jsrinnn) [reply google/cloud/bigquery/BUILD.bazel:38]: Per chat thread, Darren will take care of this. 

Resolving this comment based on above.

**Comment by jsrinnn** [comment]
Addressed all feedback and pushed all changes including install script

**Review by dbolduc — APPROVED** [review]

**Pull request merged**